### PR TITLE
feat(node): buffer isUtf8/isAscii

### PR DESCRIPTION
**Pull request by devsnek**
Fixes: https://github.com/denoland/deno/issues/23657

Implements `isUtf8` and `isAscii` as ops.

### CODE REVIEW

File: ext/node/ops/buffer.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::op2;
+use deno_core::v8;
+
+#[op2(fast)]
+pub fn op_is_ascii(buf: v8::Local<v8::ArrayBufferView>) -> bool {
```
Comment (mmastrac) [inline ext/node/ops/buffer.rs:7]: Maybe `#[arraybuffer] &[u8]` + https://doc.rust-lang.org/std/primitive.slice.html#method.is_ascii ?

Comment (nathanwhit) [inline ext/node/ops/buffer.rs:7]: You can use the `#[buffer]` attribute here and the op2 macro will do the conversion

```rust
pub fn op_is_ascii(#[buffer] buf: &[u8]) -> bool {
```

**Review by bartlomieju — APPROVED** [review]
LGTM

File: ext/node/ops/buffer.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  }
+  // SAFETY: Buffer is non-null, not detached.
+  let view = unsafe {
+    std::slice::from_raw_parts(buf.data() as *mut u8, buf.byte_length())
+  };
+  view.iter().all(|c| c.is_ascii())
+}
+
+#[op2(fast)]
+pub fn op_is_utf8(buf: v8::Local<v8::ArrayBufferView>) -> bool {
```
Comment (mmastrac) [inline ext/node/ops/buffer.rs:19]: Ditto

File: ext/node/ops/buffer.rs (hunk 1 above)
Comment (bartlomieju) [reply ext/node/ops/buffer.rs:7]: Ah, good point. I forgot we even have helpers for array buffers.

For reference: https://github.com/denoland/deno_core/tree/main/ops/op2#parameters

**Pull request merged**